src/rhub/api/_setup.py

```diff
@@ -28,11 +28,13 @@ def setup():
 
     if ADMIN_GROUP not in groups:
         logger.info(f'Creating admin group "{ADMIN_GROUP}"')
-        groups[ADMIN_GROUP] = keycloak.group_create({'name': ADMIN_GROUP})
+        keycloak.group_create({'name': ADMIN_GROUP})
+        groups = {group['name']: group for group in keycloak.group_list()}
```

Comment:
Maybe not a big deal, but... Instead of re-fetching all groups from keycloak after adding one group, you could fetch only the new group. `group_create` returns group ID you can use to call `group_get`. Same for roles, but `role_get` gets the role by name (not ID).

Reply:
Nice. I think your approach is better. Luckily, `role_create` returns the `name`, so we can do the same way.

Reply:
Just re-pushed with better variable names...